compiler/quilt/tools/build.py

```diff
@@ -80,13 +80,19 @@ def _is_valid_group(group):
     return isinstance(group, dict) or group is None
 
 def _pythonize_name(name):
-    safename = re.sub('[^A-Za-z0-9]+', '_', name).strip('_')
-
-    if safename and safename[0].isdigit():
-        safename = "n%s" % safename
+    """Convert `name` to a Python Identifier, but raise BuildException on failure."""
+    try:
+        safename = to_identifier(name, strip_underscores=True)
+    except ValueError as error:
+        raise BuildException("Invalid name: " + str(error))
+    return safename
 
-    if not is_nodename(safename):
-        raise BuildException("Unable to determine a Python-legal name for %r" % name)
+def to_nodename_reraise(name, invalid=None):
```

Comment:
This one still bothers me... We shouldn't have to wrap our own helper functions just to raise a different exception. Maybe we should add a generic exception that we catch in `main`?
(@kevinemoore, @akarve - any thoughts?)

Reply:
We could put a `QuiltException` in `constants.py` and have it be a superclass of `BuildException` and `CommandException`.  ..then `main.py` catches the `QuiltException` rather than `CommandException` as it does now.

If we want to take that further, we could say:
* File `X` raises `XException`s
* `XException`s are all `QuiltExceptions`
* The first argument to `QuiltException` should be a user-friendly (as possible) message.
* `main.py` catches `QuiltException` and displays the message (instead of doing so with `CommandException`.

Reply:
Sounds reasonable to me.

Reply:
To clarify I mean the three subclasses of `QuiltException` are fine for now. I don't think we need the `XExceptions`.

Reply:
Although, that may be better done separately -- I could do it as a PR just following the merge of this and Export into master.

Reply:
Sounds good.  I'll include these changes in this PR, and use a `QuiltException` for exceptions coming out of `util.py`.

Reply:
@dimaryaz @akarve Ready for re-review.